### PR TITLE
Start pushing prow images to gcr.io/k8s-prow-edge as well.

### PR DESCRIPTION
**Pull request by cjwagner**
See migration plan here: https://docs.google.com/document/d/1kBo24FbmeF-h1dk78wi9vnKXU-QkCXI9Ul1XEW9qVls/edit#heading=h.23gq093vwv2w

I'm setting up the edge GCR repo and adjusting credentials for the `pusher-service-account` secret's service account now.
/hold
/assign @fejta

### CODE REVIEW

File: prow/def.bzl [hunk 1]
```diff
@@ -95,11 +103,22 @@ def target(cmd):
...
+#     "gcr.io/k8s-prow-edge/plank:latest": "//prow/cmd/plank:image",
+#     "gcr.io/k8s-prow-edge/plank:latest-fejta": "//prow/cmd/plank:image",
+#     "gcr.io/k8s-prow-edge/ghproxy:20180203-deadbeef": "//ghproxy:image",
+#     "gcr.io/k8s-prow-edge/ghproxy:latest": "//ghproxy:image",
+#     "gcr.io/k8s-prow-edge/ghproxy:latest-fejta": "//ghproxy:image",
 #   }
 def tags(cmds, targets):
     # Create :YYYYmmdd-commitish :latest :latest-USER tags
     cmd_targets = {prefix(cmd): target(cmd) for cmd in cmds}
+    cmd_targets.update({edge_prefix(cmd): target(cmd) for cmd in cmds})
```
Comment (fejta) [inline prow/def.bzl]: it would be useful to only push the edge targets if edge is defined.

Comment (cjwagner) [reply prow/def.bzl]: Updated.

File: prow/def.bzl [hunk 2]
```diff
@@ -95,11 +103,23 @@ def target(cmd):
...
+#     "gcr.io/k8s-prow-edge/plank:latest-fejta": "//prow/cmd/plank:image",
+#     "gcr.io/k8s-prow-edge/ghproxy:20180203-deadbeef": "//ghproxy:image",
+#     "gcr.io/k8s-prow-edge/ghproxy:latest": "//ghproxy:image",
+#     "gcr.io/k8s-prow-edge/ghproxy:latest-fejta": "//ghproxy:image",
 #   }
 def tags(cmds, targets):
     # Create :YYYYmmdd-commitish :latest :latest-USER tags
     cmd_targets = {prefix(cmd): target(cmd) for cmd in cmds}
     cmd_targets.update({prefix(p): t for (p, t) in targets.items()})
+    if "{EDGE_PROW_REPO}":
```
Comment (fejta) [inline prow/def.bzl]: Don't you think this would be easier to understand if we set EDGE_PROW_REPO = "{EDGE_PROW_REPO}" on ~L66?

Comment (cjwagner) [reply prow/def.bzl]: /shrug
Seems about as readable to me, I was following the pattern used for `STABLE_PROW_REPO`. Updated though.

**Comment by cjwagner** [comment]
/hold cancel
PTAL

**Comment by cjwagner** [comment]
The [push job](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-test-infra-push-prow/1181272872564297735) succeeded and images are publicly readable at [gcr.io/k8s-prow-edge](https://gcr.io/k8s-prow-edge).

**Pull request merged**